File: applications/account-jdbc-sink/src/main/java/com/vmware/account/jdbc/sink/consumer/AccountConsumer.java
package com.vmware.account.jdbc.sink.consumer;

import com.vmware.account.jdbc.sink.domain.AccountEntity;
import com.vmware.account.jdbc.sink.repository.AccountRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.util.function.Consumer;

    @Component
    @RequiredArgsConstructor
    public class AccountConsumer implements Consumer<AccountEntity> {

        private final AccountRepository repository;
        @Override
        public void accept(AccountEntity account) {
            repository.save(account);

        }
    }

File: applications/stream-account-jdbc-sink/src/main/kotlin/com/vmware/tanzu/data/services/rabbitmq/streaming/account/consumer/AccountConsumer.kt
package com.vmware.tanzu.data.services.rabbitmq.streaming.account.consumer

import com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain.Account
import com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain.AccountEntity
import com.vmware.tanzu.data.services.rabbitmq.streaming.account.repository.AccountRepository
import org.springframework.stereotype.Component
import java.util.function.Consumer

/**
 * @author Gregory Green
 */
@Component
class AccountConsumer(private val accountService: AccountRepository) :
    Consumer<AccountEntity> {
    override fun accept(account: AccountEntity) {
        accountService.save(account)
    }
}
